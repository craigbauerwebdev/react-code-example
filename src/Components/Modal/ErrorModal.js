import Modal, { MODAL_TYPES } from "Components/Modal/Modal";

import LinkWrapper from "../LinkWrapper/LinkWrapper";
import ModalBody from "Components/Modal/ModalBody";
import ModalButtons from "Components/Modal/ModalButtons";
import PropTypes from "prop-types";
import React from "react";
import getAnalyticsPage from "util/getAnalyticsPage";
import modalStyles from "./scss/modal-overlay.module.scss";
import { useLocation } from "react-router";

/**
 * Error Modal with link to technical support form
 * https://github.com/Klowd/onlineeventpro-product-ui/wiki/Site-Modal
 * @param {boolean} isActive toggle display of modal
 * @param {function} onCloseErrorModal call parent component to close modal window
 * @param {string} className for styling
 */
const ErrorModal = ({
  isActive,
  onCloseErrorModal,
  customClassName,
  customMessage,
}) => {
  const location = useLocation();
  return (
    <Modal
      closeCallback={onCloseErrorModal}
      active={isActive}
      modalType={MODAL_TYPES.short}
      button={[
        <ModalButtons
          type={"confirmation"}
          key={"confirmation"}
          alignment={"center"}
          page={getAnalyticsPage(location.pathname)}
          componentType="Button"
          confirmationButtonText="Try Again"
          url="Try Again"
          componentName="Close blocked users"
        />,
        <ModalButtons type={"close-icon"} />,
      ]}
    >
      <ModalBody
        className={customClassName || modalStyles.errorModal}
        title={
          customMessage || (
            <div>
              Something went wrong. Please try again or select the{" "}
              <LinkWrapper to="/support" componentName="Support">
                Technical Support Form
              </LinkWrapper>
            </div>
          )
        }
      ></ModalBody>
    </Modal>
  );
};

ErrorModal.propTypes = {
  isActive: PropTypes.bool.isRequired,
  onCloseErrorModal: PropTypes.func.isRequired,
  customClassName: PropTypes.string,
  customMessage: PropTypes.element,
};

export default ErrorModal;
