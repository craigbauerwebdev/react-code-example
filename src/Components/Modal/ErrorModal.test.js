import ErrorModal from "./ErrorModal.js";
import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";

jest.mock("react-router", () => ({
  ...jest.requireActual("react-router"),
  useLocation: () => ({
    pathname: "mocked-path",
  }),
}));
jest.mock("./ModalButtons", () => {
  return {
    __esModule: true,
    default: () => {
      return "";
    },
  };
});
jest.mock("Components/LinkWrapper/LinkWrapper", () => {
  return {
    __esModule: true,
    default: ({ children }) => {
      return <div>{children}</div>;
    },
  };
});
it("renders without crashing", () => {
  const component = shallow(
    <ErrorModal
      isActive={true}
      onCloseErrorModal={() => null}
      className={"special class"}
    />
  );

  expect(component.find("Modal").length).toEqual(1);
  expect(toJson(component)).toMatchSnapshot();
});
