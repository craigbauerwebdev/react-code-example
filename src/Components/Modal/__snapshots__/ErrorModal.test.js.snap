// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<Modal
  active={true}
  button={
    Array [
      <default
        alignment="center"
        componentName="Close blocked users"
        componentType="Button"
        confirmationButtonText="Try Again"
        page="homepage"
        type="confirmation"
        url="Try Again"
      />,
      <default
        type="close-icon"
      />,
    ]
  }
  closeCallback={[Function]}
  modalType="short"
>
  <ModalBody
    className="errorModal"
    title={
      <div>
        Something went wrong. Please try again or select the
         
        <default
          componentName="Support"
          to="/support"
        >
          Technical Support Form
        </default>
      </div>
    }
  />
</Modal>
`;
